### PR TITLE
Update SQLite to 3.43.0 (JSON optimizations) - Second try

### PR DESCRIPTION
**Pull request by aldo-expensify**
Second try at: https://github.com/Expensify/Bedrock/pull/1555

### Details

This needs: https://github.com/Expensify/Auth/pull/8501
SQLite team added some optimizations that improve performance when handling big JSONs (for example, big policies)
This is in the context of optimizing the `SharePolicy` command.
More info: see email chain with subject "Reusing parsed JSON in indexes"

SO about updating SQLite version: https://stackoverflowteams.com/c/expensify/questions/154

How I produced the sqlite.c and sqlite.h:

1. Downloaded tar ball from https://sqlite.org/src/info/bedrock (48ecf390) as pointed out by Richard (latest update 2023-09-09 15:30)
2. Configured with `./configure --enable-fts5 --enable-update-limit`
3. Compiled with `make sqlite3.c shell.c`

Configuration/compile instructions obtained from:  https://github.com/Expensify/Salt/blob/main/build/files/jenkins/build_scripts/build_sqlite3-cli.sh#L73-L77

### Fixed Issues
For https://github.com/Expensify/Expensify/issues/289995 

### Tests

Run locally `RefundExpensifyCardSettlement` and `Billing_ExpensifyOrgDonations` and confirm that they pass
Automated tests should run and not explode

_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

**Comment by aldo-expensify** [comment]
I'm fixing some Auth queries so they work with the breaking changes of the new sqlite version: https://github.com/Expensify/Auth/pull/8501

**Comment by aldo-expensify** [comment]
https://github.com/Expensify/Auth/pull/8501 was deployed
I ran all Auth tests locally, and they passed
Asked here if we should proceed with this update or wait for a release in September: https://expensify.slack.com/archives/C03TQ48KC/p1692227588821289

**Comment by aldo-expensify** [comment]
Making this a Draft because Bondy made me notice I didn't check other repos for queries that may need fixing.

**Comment by bondydaa** [comment]
i'm mostly OOO this week so going to remove myself but please have @coleaeason or @flodnv review this as well.

**Comment by aldo-expensify** [comment]
I did a text search across all our repos looking for occurrences of `JSON_VALID`. I didn't spot any case that needed fixing.

Making this ready for review again.

**Comment by aldo-expensify** [comment]
Updated to version: https://sqlite.org/src/info/bedrock (4d8df0c426b8ce3db6cfb71e23f752026ef886b9bb833dc4be9717db9955b1db)

Re-running tests locally...

**Comment by aldo-expensify** [comment]
Tests passed

**Review by hayata-suenaga — APPROVED** [review]

**Review by flodnv — DISMISSED** [review]

**Comment by flodnv** [comment]
Adding a HOLD on https://expensify.slack.com/archives/C03TQ48KC/p1692688797520899?thread_ts=1692227588.821289&cid=C03TQ48KC

**Review by hayata-suenaga — APPROVED** [review]
I don't have much context of the SQLite configuration. I'll defer to @flodnv's approval 🙇

File: libstuff/sqlite3ext.h
Comment (aldo-expensify) [inline libstuff/sqlite3ext.h:322]: @tylerkaraszewski In the [update SQLite instructions](https://stackoverflowteams.com/c/expensify/questions/154) this file `sqlite3ext.h` is not mentioned, but I added it because there were changes. Do you know if this is fine or if I should remove it from the PR?

**Review by flodnv — APPROVED** [review]
@tylerkaraszewski please merge this PR if you're satisfied 😄

**Review by tylerkaraszewski — APPROVED** [review]

**Pull request merged**
